Lib/fnmatch.py

```diff
@@ -77,23 +77,32 @@ def translate(pat):
     There is no way to quote meta-characters.
     """
 
-    STAR = object()
-    parts = _translate(pat, STAR, '.')
-    return _join_translated_parts(parts, STAR)
+    parts, indices = _translate(pat, '*', '.')
+    return _join_translated_parts(parts, indices)
 
+_re_setops_sub = re.compile(r'([&~|])').sub
+_re_escape = functools.lru_cache(maxsize=32768)(re.escape)
 
 def _translate(pat, STAR, QUESTION_MARK):
```

Comment:
```suggestion
def _translate(pat, star, question_mark):
```

Now that `STAR` is no longer `object()`, this naming is vestigial.

Reply:
The thing is, it's still possible for other parts of the library to use different values for `STAR`. I'd like to keep it that way (e.g., `results.extend(fnmatch._translate(part, f'{not_sep}*', not_sep)[0])` in glob.py, the star is not just a "star", it's a bit more than that I'd say).

Reply:
I think this was more about making argument names lower case than anything else.

Reply:
Mmh. I like the fact that it's all caps to highlight the fact that it's a constant, as if `_translate` was a macro in this case. But maybe it's better to drop this historical approach and move to the modern one.

Reply:
The thing is, `STAR` _used_ to be a module-level constant in `fnmatch`, but I changed that when I introduced `glob.translate()`. I should have changed them to lower case in that PR, and I regret not doing so now.

Reply:
I see. I hope the last commit solved your regrets :)